cache/class-vary-cache.php

```diff
@@ -426,11 +426,17 @@ private static function parse_group_cookie() {
 	 * @return string A string representation of the groups
 	 */
 	private static function stringify_groups() {
-		ksort( self::$groups ); // make sure the string order is the same every time.
+		$groups = array_filter(
+			self::$groups,
+			function( $value ) {
+				return '' !== $value;
+			}
+		);
+		ksort( $groups ); // make sure the string order is the same every time.
 		$flatten = function ( $key, $value ) {
 			return $key . self::VALUE_SEPARATOR . $value;
 		};
-		$flattened = array_map( $flatten, array_keys( self::$groups ), self::$groups );
+		$flattened = array_map( $flatten, array_keys( $groups ), $groups );
```

Comment:
I wonder if a `foreach` would be simpler here?

```
$flattened = [];
foreach ( self::$groups as $key => $value ) {
    if ( empty( $value ) ) {
        continue;
    }

    $flattened[] = $key . self::VALUE_SEPARATOR . $value;
}
```

Reply:
Hmm, I'm on the fence if that makes it simpler or not.
Perhaps a comment explaining the array_map options here?
I feel like there's a way to do it with an array_merge and the spread operator but that might be even more obscure.

Reply:
With the foreach, we have one loop. With the array_map and array_filter we have two. 

Plus there's this: https://stackoverflow.com/questions/18144782/performance-of-foreach-array-map-with-lambda-and-array-map-with-static-function

Obviously we're looking at really tiny arrays so the performance costs with be mostly negligable but still worth optimizing where we can.

Reply:
Okay, I'm convinced :)